components/Form/FormField.jsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import AutoComplete from 'material-ui/AutoComplete';
+import ChipList from './ChipList';
+import { connect } from 'react-redux';
+import { addChip, setFieldText } from '../../actions';
+import { form } from '../../reducers';
+import { getClosestMatch } from './utility';
+import styles from './FormField.css';
+
+const onNewRequest = (value, index, formName, name, dataSourceConfig, dataSource, strict, addChip, setFieldText, limit) => {
```

Comment:
yea this sickens me. There are several options I can see here.

1. Keep as is. Hard to read the method signature and easy to mess up.
2. Pass an object. Still hard to read but can't mess up signature. Incurs object creation overhead (minor)
3. Put the function into the stateless component and don't pass anything into the component. No arguments then (inheriting from function scope), but confusing to understand where all the variables come from.
4. Hopefully something magically better than these three that you or I think up.

I will convert this to the destructuring for now but, let's try to find a better way to do this.